test/tasks/redo_task_test.rb

```diff
@@ -0,0 +1,152 @@
+require_relative './task_test_base'
+
+module Mongoid
+  class RedoTaskTest < TaskTestBase
+    def test_database_migrate_redo
+      Mongoid::Migrator.migrations_path = [MIGRATIONS_ROOT + "/valid"]
+      invoke("db:migrate")
+      assert_output(<<-EOF
+
+database: mongoid_test
+
+ Status   Migration ID    Migration Name
+--------------------------------------------------
+   up     20100513054656  AddBaselineSurveySchema
+   up     20100513055502  AddSecondSurveySchema
+   up     20100513063902  AddImprovementPlanSurveySchema
+EOF
+      ) { invoke("db:migrate:status") }
+      invoke("db:migrate:redo")
+      assert_output(<<-EOF
+
+database: mongoid_test
+
+ Status   Migration ID    Migration Name
+--------------------------------------------------
+   up     20100513054656  AddBaselineSurveySchema
+   up     20100513055502  AddSecondSurveySchema
+   up     20100513063902  AddImprovementPlanSurveySchema
+EOF
+      ) { invoke("db:migrate:status") }
+    end
+
+    def test_database_migrate_redo_version
+      Mongoid::Migrator.migrations_path = [MIGRATIONS_ROOT + "/valid"]
+      invoke("db:migrate")
+      assert_output(<<-EOF
+
+database: mongoid_test
+
+ Status   Migration ID    Migration Name
+--------------------------------------------------
+   up     20100513054656  AddBaselineSurveySchema
+   up     20100513055502  AddSecondSurveySchema
+   up     20100513063902  AddImprovementPlanSurveySchema
+EOF
+      ) { invoke("db:migrate:status") }
+      with_env("VERSION" => "20100513055502") { invoke("db:migrate:redo") }
+      assert_output(<<-EOF
+
+database: mongoid_test
+
+ Status   Migration ID    Migration Name
+--------------------------------------------------
+   up     20100513054656  AddBaselineSurveySchema
+   up     20100513055502  AddSecondSurveySchema
+   up     20100513063902  AddImprovementPlanSurveySchema
+EOF
+      ) { invoke("db:migrate:status") }
```

Comment:
There was no `down` migration before the redo, so there's no proof that the redo with the version is not targeting the last version (same for the next specs).